src/csvcubed/utils/cache.py

```diff
@@ -1,4 +1,197 @@
+    """
+
+    templates_dir = APP_ROOT_DIR_PATH / "readers" / "cubeconfig" / "v1_0" / "templates"
+
+    template_files = templates_dir.rglob("**/*.json*")
+
+    if not any(template_files):
+        raise ValueError(f"Couldn't find template files in {templates_dir}.")
+
+    map_uri_to_file_path = copy.deepcopy(_hard_codes_map_url_to_file_path)
+
+    for template_file in template_files:
+        relative_file_path = str(template_file.relative_to(templates_dir))
+        uri = (
+            "//raw.githubusercontent.com/GSS-Cogs/csvcubed/main/src/csvcubed/readers/cubeconfig/v1_0/templates/"
+            + relative_file_path
+        )
+        map_uri_to_file_path[uri] = template_file
+
+    return map_uri_to_file_path
+
+
+map_url_to_file_path = _get_url_to_file_path_map()
+
+
+class CustomAdapterServeSomeFilesLocally(BaseAdapter):
+    http_adapter: HTTPAdapter
+
+    def __init__(self):
+        self.http_adapter = HTTPAdapter()
+
+    def send(
+        self, request, stream=False, timeout=None, verify=True, cert=None, proxies=None
+    ):
+        _logger.debug(
+            "This is the HTTP(s) adapter sending the request: %s", request.url
+        )
+        if request.url is None:
+            raise requests.exceptions.URLRequired
+        try:
+            response = self.http_adapter.send(
+                request,
+                stream=stream,
+                timeout=timeout,
+                verify=verify,
+                cert=cert,
+                proxies=proxies,
+            )
+
+        except requests.exceptions.ConnectionError as e:
+            try:
+                _logger.debug("The connection error has been found")
+
+                path_to_local_file = generate_path_to_local_file(request.url)
+
+                _logger.debug("The local file path is: %s", path_to_local_file)
+
+                return create_local_copy_response(path_to_local_file, request)
+
+            except KeyError:
+                raise FileNotFoundError(
+                    f"URL {request.url} did not produce a response and a local copy could not be found at the corresponding mapped path."
+                ) from e
+
+        if response.status_code >= 400 and response.status_code < 600:
+            _logger.info(f"The status code is {response.status_code}")
+            try:
+                path_to_local_file = generate_path_to_local_file(request.url)
+                return create_local_copy_response(path_to_local_file, request, response)
+
+            except KeyError:
+                raise FileNotFoundError(
+                    f"URL {request.url} produced a invalid response and a local copy could not be found at the corresponding mapped path."
+                )
+
+
+        return response
+
+    def close(self) -> None:
+        self.http_adapter.close()
+
+
+def generate_path_to_local_file(request_url: str) -> Path:
+    trimmed_url = str(request_url).removeprefix("https:")
```

Comment:
Will this handle `http://` URLs as well?

Reply:
added a removeprefix for `http:` and removed the logic to handle the trailing forward slash